### ISSUE TITLE
Error occurred when creating type stub for sqlglot on first try

### ISSUE THREAD

**Issue by Hunterlige** [microsoft/pyright#8235]
**Describe the bug**
Createstub fails with `sqlglot` on the first try. If the command is run a second time, the command will succeed. This is a regression from pyright that happens in `1.1.367`. `1.1.366` works well on first try with the same `sqlglot` version.

**Code or Screenshots**
```sh
pip install sqlglot==22.5.0
pyright --createstub sqlglot
>>> Error occurred when creating type stub: Debug Failure. False expression.
pyright --createstub sqlglot
>>> Type stub was created for 'sqlglot'
```

Pyright CLI version: 1.1.369

**Comment by erictraut** [comment]
While it's not immediately obvious, this turns out to be a duplicate of https://github.com/microsoft/pyright/issues/8228. I'll leave both open until the issue is fixed.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.370
